nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -253,9 +253,13 @@ def _build_dataset(self, data_cfg, is_train=True):
 
         # TE requires that the first input dim is divisible by 8 and the second by 16 for fp8
         # When using sequence parallel, sequence will further be split by TP size
-        pad_seq_length_to_mult = (
-            8 * self.cfg.get('tensor_model_parallel_size', 1) if self.cfg.get('sequence_parallel', False) else 16
-        )
```

Comment:
@vysarge I think you possibly only need to add  `pad_seq_length_to_mult *= self.cfg.get('context_parallel_size', 1)` at here?